### PR TITLE
mgmt/osdp: Fix compilation when CONFIG_OSDP_SC_ENABLED enabled

### LINKED ISSUE #66289: OSDP Samples Minor Bugs
**Bug Description**
When Building a project for the **nrf52840dk_nrf52840** board I encountered some minor bugs that caused the build to fail.
The first was a missing **'t'** in a variable declaration with type **uint8_t**. The second was a missing opening curly brace for a function.

**Platform & Context**
I am using:
- Zephyr Toolchain 0.16.4
- nrf SDK 2.4.99-dev2
- nrf52840dk_nrf52840 board

**Error Messages**
When building the **peripheral_device** sample I encounter these errors:
`/opt/nordic/ncs/v2.4.99-dev2/zephyr/subsys/mgmt/osdp/src/osdp_sc.c:28:18: error: unknown type name 'uin8_t'
   28 |         volatile uin8_t *p = mem;`
   
  and

`/opt/nordic/ncs/v2.4.99-dev2/zephyr/subsys/mgmt/osdp/src/osdp_phy.c:224:25: error: label 'out_of_space_error' used but not defined
  224 |                         goto out_of_space_error;` 
  
  
  **My Solutions**
  Investigation of the errors led me to these files:
- osdp_sc.c  
- osdp_phy.c

In **osdp_sc.c**, on line _28_  in the _osdp_memzero_ function:
`volatile uin8_t *p = mem;`. was changed to `volatile uint8_t *p = mem;`

In **osdp_phy.c**, on line _197_ in the _osdp_phy_packet_finalize_:
`(pkt->control & PKT_CONTROL_SCB) && pkt->data[1] >= SCS_15)` was changed to `(pkt->control & PKT_CONTROL_SCB) && pkt->data[1] >= SCS_15) {`

Following those minor additions to the files, the OSDP sample application "peripheral-device" built successfully

### PR DESCRIPTION
**Pull request by arkwad**
The 2 issues were found after CONFIG_OSDP_SC_ENABLED was enabled:
- missing curly brace after if
- typo in type name

### CODE REVIEW

File: subsys/mgmt/osdp/src/osdp_phy.c [hunk 1]
```diff
@@ -233,7 +233,6 @@ int osdp_phy_packet_finalize(struct osdp_pd *pd, uint8_t *buf,
 		data = is_cp_mode(pd) ? pd->sc.c_mac : pd->sc.r_mac;
 		memcpy(buf + len, data, 4);
 		len += 4;
-	}
```
Comment (kartben) [inline subsys/mgmt/osdp/src/osdp_phy.c:236]: looking at the indenting it looks like it's rather a missing opening curly brace at line 197?

**Review by kartben — CHANGES_REQUESTED** [review]

**Comment by jakub-uC** [comment]
@arkwad
As @kartben mentioned it would be good to have a sample compiled by the CI.

You can create a prj.conf that will have a secure channel active. Then, in the testcase yaml, add it as compile only. You can refer to the shell tests for guidance:

File: subsys/mgmt/osdp/src/osdp_phy.c (hunk 1 above)
Comment (arkwad) [reply subsys/mgmt/osdp/src/osdp_phy.c]: you're right! fixed that :)

**Review by jakub-uC — DISMISSED** [review]

**Review by kartben — CHANGES_REQUESTED** [review]

File: samples/subsys/mgmt/osdp/control_panel/sample.yaml [hunk 2]
```diff
@@ -10,3 +10,7 @@ tests:
     harness: osdp
     integration_platforms:
       - stm32_min_dev_black
+  sample.mgmt.osdp.control_panel:
```
[the remark this answers is not in the document]
Comment (arkwad) [reply samples/subsys/mgmt/osdp/control_panel/sample.yaml]: ahh, my bad - fixed :)

**Review by r2r0 — DISMISSED** [review]

**Review by kartben — CHANGES_REQUESTED** [review]

File: samples/subsys/mgmt/osdp/control_panel/sample.yaml [hunk 3]
```diff
@@ -10,3 +10,7 @@ tests:
     harness: osdp
     integration_platforms:
       - stm32_min_dev_black
+  sample.mgmt.osdp.control_panel_sc:
+    build_only: true
+    extra_args: CONF_FILE=prj_sc.conf
+    tags: osdp
```
[the remark this answers is not in the document]
Comment (arkwad) [reply samples/subsys/mgmt/osdp/control_panel/sample.yaml]: Done! :) tests passed :)

**Review by r2r0 — APPROVED** [review]

**Review by kartben — APPROVED** [review]

**Review by sidcha — APPROVED** [review]

**Pull request merged**